pkg/kv/kvserver/replica_consistency.go

```diff
@@ -621,6 +620,59 @@ func (r *Replica) sha512(
 		return err
 	}
 
+	rangeKeyVisitor := func(unsafeKey storage.MVCCRangeKey, unsafeValue []byte) error {
+		// Rate Limit the scan through the range
+		err := limiter.WaitN(ctx, int64(len(unsafeKey.StartKey)+len(unsafeKey.EndKey)+len(unsafeValue)))
+		if err != nil {
+			return err
+		}
+
+		if snapshot != nil {
+			// Add (a copy of) the range kv pair into the debug message.
+			rkv := roachpb.RaftSnapshotData_RangeKeyValue{
+				Timestamp: unsafeKey.Timestamp,
+			}
+			alloc, rkv.StartKey = alloc.Copy(unsafeKey.StartKey, 0)
+			alloc, rkv.EndKey = alloc.Copy(unsafeKey.EndKey, 0)
+			alloc, rkv.Value = alloc.Copy(unsafeValue, 0)
+			snapshot.RangeKV = append(snapshot.RangeKV, rkv)
+		}
+
+		// Encode the length of the start key, end key, and value.
+		binary.LittleEndian.PutUint64(intBuf[:], uint64(len(unsafeKey.StartKey)))
```

Comment:
I know we do that for point keys too, but could you remind me why?

Also, what does `intBuf[:]` do again? It seems identical to `intBuf`: https://go.dev/play/p/QjMUqx3Z8Ks

Reply:
>I know we do that for point keys too, but could you remind me why?

No idea. I'm guessing it was maybe lifted from somewhere that we do KV encoding that needs length prefixes for decoding, e.g. `MVCCScanDecodeKeyValue`.

I figured I'd just keep it consistent here, but happy to drop the length hashing too.

> Also, what does `intBuf[:]` do again?

`intBuf` is an `[8]byte` array, `PutUInt64()` takes a `[]byte` slice. `[:]` casts an array to a slice.